packages/docprovider/src/ydrive.ts

```diff
@@ -57,10 +60,57 @@ export class YDrive extends Drive implements ICollaborativeDrive {
    */
   readonly sharedModelFactory: ISharedModelFactory;
 
+  disposeTimelineWidget(): void {
+    if (this._timelineWidget) {
+      this._timelineWidget.dispose();
+      this._timelineWidget = null;
+    }
+  }
+
+  setTimelineWidget(widget: TimelineWidget) {
+    this._timelineWidget = widget;
+  }
+
+  getTimelineWidget(): TimelineWidget | null {
+    return this._timelineWidget || null;
+  }
+
+  async updateTimelineForNotebook(notebookPath: string): Promise<void> {
```

Comment:
It looks like it supports not only notebook but any documents, right? Later below I propose to rewrite this procedure with side effects into a more generic function, let's call it `getProviderFor(path: string): Promise<WebSocketProvider>`